miden-crypto/src/merkle/store/mod.rs

```diff
@@ -93,22 +89,22 @@ pub struct StoreNode {
 /// ```
 #[derive(Debug, Clone, Eq, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct MerkleStore<T: KvMap<Word, StoreNode> = BTreeMap<Word, StoreNode>> {
-    nodes: T,
+pub struct MerkleStore {
+    nodes: HashMap<Word, StoreNode>,
 }
 
-impl<T: KvMap<Word, StoreNode>> Default for MerkleStore<T> {
+impl Default for MerkleStore {
```

Comment:
Can we `Derive` this?

Reply:
I don't think so because the intention seems to be to have the default state populated with empty hashes:
```rust
    /// Creates an empty `MerkleStore` instance.
    pub fn new() -> MerkleStore {
        // pre-populate the store with the empty hashes
        let nodes = empty_hashes().into_iter().collect();
        MerkleStore { nodes }
    }
```